solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -163,14 +145,11 @@ private ClusterState fetchClusterState(
     }
 
     for (Map.Entry<String, Object> e : collectionsMap.entrySet()) {
-      @SuppressWarnings("rawtypes")
-      Map m = (Map) e.getValue();
-      Long creationTimeMillisFromClusterStatus = (Long) m.get("creationTimeMillis");
-      Instant creationTime =
-          creationTimeMillisFromClusterStatus == null
-              ? Instant.EPOCH
-              : Instant.ofEpochMilli(creationTimeMillisFromClusterStatus);
-      cs = cs.copyWith(e.getKey(), fillPrs(znodeVersion, e, creationTime, m));
+      String collectionName = e.getKey();
+      Map<String, Object> collStateMap = (Map<String, Object>) e.getValue();
+      cs =
+          cs.copyWith(
```

Comment:
Calling copyWith in a loop of collections is `O(N^2)` (N=#collections); right?  Not a problem introduced in this PR but I just want to confer with you on this problem.

Reply:
Yes it is indeed O(N^2).